tools/image-info

```diff
@@ -396,31 +668,989 @@ def read_rhsm(tree):
+            "runcmd"
+        ],
+        "cloud_final_modules": [
+            "rightscale_userdata",
+            "scripts-per-once",
+            "scripts-per-boot",
+            "scripts-per-instance",
+            "scripts-user",
+            "ssh-authkey-fingerprints",
+            "keys-to-console",
+            "phone-home",
+            "final-message",
+            "power-state-change"
+        ],
+        "cloud_init_modules": [
+            "disk_setup",
+            "migrator",
+            "bootcmd",
+            "write-files",
+            "growpart",
+            "resizefs",
+            "set_hostname",
+            "update_hostname",
+            "update_etc_hosts",
+            "rsyslog",
+            "users-groups",
+            "ssh"
+        ],
+        "disable_root": 1,
+        "disable_vmware_customization": false,
+        "mount_default_fields": [
+            null,
+            null,
+            "auto",
+            "defaults,nofail,x-systemd.requires=cloud-init.service",
+            "0",
+            "2"
+        ],
+        "resize_rootfs_tmp": "/dev",
+        "ssh_deletekeys": 1,
+        "ssh_genkeytypes": null,
+        "ssh_pwauth": 0,
+        "syslog_fix_perms": null,
+        "system_info": {
+        "default_user": {
+            "gecos": "Cloud User",
+            "groups": [
+                "adm",
+                "systemd-journal"
+            ],
+            "lock_passwd": true,
+            "name": "ec2-user",
+            "shell": "/bin/bash",
+            "sudo": [
+                "ALL=(ALL) NOPASSWD:ALL"
+            ]
+        },
+        "distro": "rhel",
+        "paths": {
+            "cloud_dir": "/var/lib/cloud",
+            "templates_dir": "/etc/cloud/templates"
+        },
+        "ssh_svcname": "sshd"
+        },
+        "users": [
+            "default"
+        ]
+    }
+    """
+    result = {}
+
+    with contextlib.suppress(FileNotFoundError):
+        with open(f"{tree}/etc/cloud/cloud.cfg") as f:
+            config = yaml.safe_load(f)
+            result.update(config)
+
+    return result
+
+
+def read_dracut_conf_d(tree):
+    """
+    Read all *.conf files from /etc/dracut.conf.d/.
+
+    Returns: dictionary with the keys representing names of configuration files
+    from /etc/dracut.conf.d. Value of each key is a dictionary representing the
+    uncommented configuration options read from the file.
+
+    An example return value:
+    {
+        "sgdisk.conf": {
+            "install_items": " sgdisk "
+        },
+        "xen.conf": {
+            "add_drivers": " xen-netfront xen-blkfront "
+        }
+    }
+    """
+    result = {}
+
+    # iterate through all *.conf files in /etc/dracut.conf.d/
+    files = glob.glob(f"{tree}/etc/dracut.conf.d/*.conf")
+    for file in files:
+        confname = os.path.basename(file)
+        config = {}
+        with open(file) as f:
+            # dracut configuration key/values delimiter is '=' or '+='
```

Comment:
Based on what I read in `dracut.conf(5)`, this is really tied to the option. So the answer if it has different semantics is yes, "kind of". Options that add additional values to s list of already existing ones tend to use `+=` instead of `=` for the assignment. The usage is specific to the option name and you can not use both (AFAIK). So while I understand the implementation choice to make it explicit which options are "adding" values to some list, from the ones that are assigning the provided value. But it didn't seem to be that important for `image-info` use case.

Reply:
Thanks for the explanation!